ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -0,0 +1,3018 @@
+  ///
+  /// The key format is $languageID_$keyboardID. For example, "eng_european2" returns the English EuroLatin2 keyboard
+  public private(set) var keyboardsDictionary: [String: [String: String]] = [:]
+
+  /// Dictionary of available Keyman keyboard fonts
+  public private(set) var keymanFonts: [String: [String: Any]] = [:]
+
+  /// Keyman system-wide keyboard
+  public let isSystemKeyboard: Bool
+
+  /// The version of the Keyman SDK
+  public var sdkVersion: String {
+    let info = NSDictionary(contentsOfFile: keymanBundle.path(forResource: "KeymanEngine-Info",
+                                                              ofType: "plist")!)
+    return info!["CFBundleVersion"] as! String
+  }
+
+  /// Keyman Web resources
+  public var keymanBundle: Bundle {
+    return Bundle(path: Bundle(for: Manager.self).path(forResource: "Keyman", ofType: "bundle")!)!
+  }
+
+  /// In keyboard extensions (system keyboard), `UIApplication.openURL(_:)` is unavailable. The API is not called in
+  /// the system keyboard since `KeyboardInfoViewController` is never used. `openURL(:_)` is only used in applications,
+  /// where it is safe. However, the entire Keyman Engine framework must be compiled with extension-safe APIs.
+  ///
+  /// Set this to `UIApplication.shared.openURL` in your application.
+  public var openURL: ((URL) -> Bool)?
+
+  var keyboardID: String?
+  var languageID: String?
+  weak var webDelegate: KeymanWebViewDelegate?
+  weak var inputDelegate: KeymanWebViewDelegate?
+  var currentRequest: HTTPDownloadRequest?
+  var keyboardsInfo: [String: [String: String]]?
+  var shouldReloadKeyboard = false
+  var inputView: WKWebView! = nil
+
+  private var downloadQueue: HTTPDownloader?
+  private var sharedQueue: HTTPDownloader!
+  private var reachability: Reachability!
+  private var didSynchronize = false
+  private var didResizeToOrientation = false
+  private var lastKeyboardSize: CGSize = .zero
+  private var specialOSKFont: String?
+
+  private let subKeyColor = #colorLiteral(red: 244.0 / 255.0, green: 244.0 / 255.0, blue: 244.0 / 255.0, alpha: 1.0)
+  private let subKeyColorHighlighted = #colorLiteral(red: 136.0 / 255.0, green: 136.0 / 255.0, blue: 1.0, alpha: 1.0)
+
+  // Views
+  private var helpBubbleView: PopoverView?
+  private var keyPreviewView: KeyPreviewView?
+  private var subKeysView: SubKeysView?
+  private var keyboardMenuView: KeyboardMenuView?
+
+  // Arrays
+  private var subKeyIDs: [String] = []
+  private var subKeyTexts: [String] = []
+  private var subKeys: [UIButton] = []
+
+  // Key frames
+  private var keyFrame = CGRect.zero
+  private var menuKeyFrame = CGRect.zero
+
+  // Dictionary of Keyman options
+  var options: [AnyHashable: Any] = [:]
+
+  // MARK: - Object Admin
+  deinit {
+    NotificationCenter.default.removeObserver(self)
+    if currentRequest?.userInfo != nil {
+      currentRequest!.userInfo["completionBlock"] = nil
+    }
+  }
```

Comment:
Yes it's unnecessary and should have been unnecessary before. If an object is being deallocated, it'll release the references it holds anyways. Plus since `Manager` is a singleton, it shouldn't be deallocated while the app is running.